src/qobj/quantum_object.jl

```diff
@@ -213,58 +213,64 @@ function QuantumObject(
         elseif type isa OperatorBraQuantumObject
             dims = [isqrt(Size[2])]
         end
+    else
+        _check_dims(dims)
     end
-
-    _check_QuantumObject(type, prod(dims), Size[1], Size[2])
+    _check_QuantumObject(type, dims, Size[1], Size[2])
     return QuantumObject(A, type, dims)
 end
 
-function _check_QuantumObject(type::KetQuantumObject, prod_dims::Int, m::Int, n::Int)
-    (n != 1) ? throw(DomainError((m, n), "The dimension of the array is not compatible with Ket type")) : nothing
-    return prod_dims != m ? throw(DimensionMismatch("The dims parameter does not fit the dimension of the Array.")) :
-           nothing
+_check_dims(dims::Vector{Int}) =
+    all(>(0), dims) ? nothing : throw(DomainError(dims, "The argument dims must be a vector with positive integers."))
```

Comment:
you can write `all(>(0), dims) || throw(…`

Reply:
ahh, yes
thank you ~

Reply:
okay, done~